config/initializers/devise.rb

```diff
@@ -96,7 +96,7 @@
   # a value less than 10 in other environments. Note that, for bcrypt (the default
   # encryptor), the cost increases exponentially with the number of stretches (e.g.
   # a value of 20 is already extremely slow: approx. 60 seconds for 1 calculation).
-  config.stretches = Rails.env.test? ? 1 : 12
+  config.stretches = Rails.env.test? ? 1 : 100_000
```

Comment:
I think that might be too high. Ideally, we'd want to shoot for a 100ms-to-glass response time, so we should keep server-side processes as fast as possible. I think 100_000 is a good value to start with since it matches the Devise default 11 stretches for bcrypt.

The 100ms number is a guideline based on this 1968 study on [human-computer interaction speeds](http://theixdlibrary.com/pdf/Miller1968.pdf), and is still considered valid today AFAIK.

> 0.1 second is about the limit for having the user feel that the system is reacting instantaneously, meaning that no special feedback is necessary except to display the result.